extensions/dicom-segmentation/src/getSourceDisplaySet.js

```diff
@@ -36,14 +36,32 @@ const _getReferencedDisplaySet = (segDisplaySet, studies) => {
     referencedSeriesInstanceUIDs = ReferencedSeriesSequence.map(
       ReferencedSeries => ReferencedSeries.SeriesInstanceUID
     );
-  } else {
-    const { PerFrameFunctionalGroupsSequence } = metadata;
+  }
+
+  if (metadata.ReferencedImageSequence &&
+    (!referencedSeriesInstanceUIDs || referencedSeriesInstanceUIDs.length === 0)) {
```

Comment:
Can we extract this block into a separate jsDocs function? this will help understand this code block in the future